paasta_tools/setup_marathon_job.py

```diff
@@ -696,24 +698,25 @@ def log_deploy_error(errormsg: str, level: str = "event") -> None:
             new_app, draining_hosts=draining_hosts
         )
         if new_app.instances < config["instances"] + num_at_risk_tasks:
+            new_instance_count = config["instances"] + num_at_risk_tasks
+            if max_instances and max_instances < new_instance_count:
```

Comment:
If `max_instances` can be zero, then `if max_instances is not None and ...`

Reply:
Fixed.